nsxt/provider.go

```diff
@@ -407,13 +413,19 @@ func Provider() *schema.Provider {
 }
 
 func configureNsxtClient(d *schema.ResourceData, clients *nsxtClients) error {
+	onDemandConn := d.Get("on_demand_connection").(bool)
 	clientAuthCertFile := d.Get("client_auth_cert_file").(string)
 	clientAuthKeyFile := d.Get("client_auth_key_file").(string)
 	clientAuthCert := d.Get("client_auth_cert").(string)
 	clientAuthKey := d.Get("client_auth_key").(string)
 	vmcToken := d.Get("vmc_token").(string)
 	vmcAuthMode := d.Get("vmc_auth_mode").(string)
 
+	if onDemandConn {
```

Comment:
Does it make more sense if we make vmc params taking precedence over this `on_demand_connection`, in that if vmc tokens / mode is set, `on_demand_connection` is ignored and turned off?

Reply:
I think its preferrable that user is aware the setting is not applicable, and corrects the configuration